src/commands/typescript/update.ts

```diff
@@ -32,30 +37,69 @@ export default class Update extends SfdxCommand {
     }),
   };
 
-  private typescriptPkg: NpmPackage;
+  private typescriptPkgs: NpmPackage[];
 
   public async run(): Promise<void> {
-    this.typescriptPkg = this.retrieveTsPackage();
+    this.typescriptPkgs = await this.retrieveTsPackages();
     this.validateEsTarget();
     this.validateTsVersion();
 
+    const pkg = this.isLernaRepo()
+      ? await LernaRepo.create({ ux: this.ux })
+      : await SinglePackageRepo.create({ ux: this.ux });
+
     this.ux.warn('This is for testing new versions only. To update the version you must go through dev-scripts.');
+
     await this.updateTsVersion();
     await this.updateEsTarget();
 
-    const pkg = await SinglePackageRepo.create({ ux: this.ux });
-    try {
-      pkg.install();
-      pkg.build();
-      pkg.test();
-    } finally {
-      this.ux.log('Reverting unstaged stages');
-      pkg.revertUnstagedChanges();
+    if (this.isLernaRepo()) {
+      const packagePaths = await this.getPackagePaths();
+      const workingDir = pwd().stdout;
+
+      for (const packagePath of packagePaths) {
```

Comment:
There's no need to iterate over the packages since you can run `yarn install`, `yarn test` at the top level

Doing it at the top level also eliminates the need to `cd` into different working directories